infrastructure/smart-contracts/contracts/upgrades/referee/Referee9.sol

```diff
@@ -830,51 +830,70 @@ contract Referee9 is Initializable, AccessControlEnumerableUpgradeable {
     
     function stakeKeys(address pool, address staker, uint256[] memory keyIds, bool isAdminStake) external onlyPoolFactory {
         uint256 keysLength = keyIds.length;
+        // Check if the pool has enough capacity to stake the keys
         require(assignedKeysToPoolCount[pool] + keysLength <= maxKeysPerPool, "43");
 
         uint256 currentChallenge = challengeCounter - 1;
 
         NodeLicense nodeLicenseContract = NodeLicense(nodeLicenseAddress);
+        // Assign the keys to the pool
         for (uint256 i = 0; i < keysLength; i++) {
             uint256 keyId = keyIds[i];
+            // Check if the key is not already assigned to a pool
             require(assignedKeyToPool[keyId] == address(0), "44");
             if(!isAdminStake){
+                // If not admin stake, check if the staker is the owner of the key
                 require(nodeLicenseContract.ownerOf(keyId) == staker, "45");
             }
             assignedKeyToPool[keyId] = pool;
-            if(submissions[currentChallenge][keyId].submitted && submissions[currentChallenge][keyId].eligibleForPayout){
-                submissions[currentChallenge][keyId].eligibleForPayout = false;
-                challenges[currentChallenge].numberOfEligibleClaimers--;
-                emit AssertionCancelled(currentChallenge, keyId);
```

Comment:
We should remove the `AssertionCancelled` event from the Referee and from the subgraph here

Reply:
I can see the event being removed in https://github.com/xai-foundation/sentry/pull/320/files so this is resolved